lib/ansible/modules/packaging/language/pip.py

```diff
@@ -235,6 +244,95 @@
                              'pip': 'import pkg_resources; print(pkg_resources.get_distribution("pip").version)'}
 
 
+class Distribution:
+
+    """Wrapper class for pkg_resources.Requirement.
+
+    A wrapper class for Requirement, provide usefual API to parse package name and version specifier,
+    do fallback if the installed setuptools version doesn't support some features.
+
+    """
+
+    def __init__(self, name_string, version_string=None):
+        if not (name_string.startswith('file:') or _is_vcs_url(name_string)):
+            self._plain_distribution = True
+            if version_string:
+                version_string = version_string.lstrip()
+                separator = '==' if version_string[0].isdigit() else ' '
+                name_string = separator.join((name_string, version_string))
+            self._requirement = Requirement.parse(name_string)
+            # old pkg_resource will replace 'setuptools' with 'distribute' when it already installed
+            if self._requirement.project_name == "distribute":
+                self._distribution_name = "setuptools"
+            else:
+                self._distribution_name = self._requirement.project_name
+        else:
+            self._plain_distribution = False
+            self._distribution_name = name_string
+
+    @property
+    def has_version_specifier(self):
+        if self._plain_distribution:
+            return bool(self._requirement.specs)
+        return False
+
+    @property
+    def distribution_name(self):
+        return self._distribution_name
+
+    def is_satisfied_by(self, version_to_test):
+        if self._plain_distribution:
+            if hasattr(self._requirement, 'specifier'):
+                return self._requirement.specifier.contains(version_to_test)
+            else:
+                version_to_test = LooseVersion(version_to_test)
+                op_dict = {">=": operator.ge, "<=": operator.le, ">": operator.gt,
+                           "<": operator.lt, "==": operator.eq, "!=": operator.ne}
+                # old setuptools has no specifier, do fallback
+                for (op, ver) in self._requirement.specs:
+                    if not op_dict[op](version_to_test, LooseVersion(ver)):
+                        return False
+
+                return True
+        return False
+
+    def __str__(self):
+        if self._plain_distribution:
+            return str(self._requirement)
+        return self._distribution_name
+
+
+def _is_vcs_url(name):
+    """Test whether a name is a vcs url or not."""
+    return re.match(r'(svn|git|hg|bzr)\+', name)
+
+
+def _is_valid_distribution_name(name):
+    """Test whether a name is distribution name or version specifier."""
+    return not name.lstrip().startswith(('>=', '<=', '!=', '==', '>', '<'))
```

Comment:
`'>=', '<=', '!=', '==', '>', '<'` is duplicated with that operator map. you could do global map and use `that_map.keys()`